src/ompl/geometric/planners/bitstar/datastructures/src/IntegratedQueue.cpp

```diff
@@ -0,0 +1,1693 @@
+            }
+        }
+
+
+
+        void BITstar::IntegratedQueue::listVertices(std::vector<VertexConstPtr>* vertexQueue)
+        {
+            //Clear the given list:
+            vertexQueue->clear();
+
+            //Iterate until the end, pushing back:
+            for (cost_vertex_multimap_t::const_iterator vIter = vertexToExpand_; vIter != vertexQueue_.end(); ++vIter)
+            {
+                //Push back:
+                vertexQueue->push_back(vIter->second);
+            }
+        }
+
+
+
+        void BITstar::IntegratedQueue::listEdges(std::vector<std::pair<VertexConstPtr, VertexConstPtr> >* edgeQueue)
+        {
+            //Clear the vector
+            edgeQueue->clear();
+
+            //I don't think there's a std::copy way to do this, so just iterate
+            for( cost_pair_vertex_pair_multimap_t::const_iterator eIter = edgeQueue_.begin(); eIter != edgeQueue_.end(); ++eIter )
+            {
+                edgeQueue->push_back(eIter->second);
+            }
+        }
+
+
+
+
+
+
+
+
+
+        void BITstar::IntegratedQueue::updateQueue()
+        {
+            //Variables:
+            //Whether to expand:
+            bool expand;
+
+            expand = true;
+            while ( expand == true )
+            {
+                //Check if there are any vertices to expand:
+                if (vertexToExpand_ != vertexQueue_.end())
+                {
+                    //Expand a vertex if the edge queue is empty, or the vertex could place a better edge into it:
+                    if (edgeQueue_.empty() == true)
+                    {
+                        //The edge queue is empty, any edge is better than this!
+                        this->expandNextVertex();
+                    }
+                    //This is isCostBetterThanOrEquivalentTo because of the second ordering criteria. The vertex expanded could match the edge in queue on total cost, but have less cost-to-come.
+                    else if (this->isCostBetterThanOrEquivalentTo( vertexToExpand_->first, edgeQueue_.begin()->first.first ) == true)
+                    {
+                        //The vertex *could* give a better edge than our current best edge:
+                        this->expandNextVertex();
+                    }
+                    else
+                    {
+                        //We are done expanding for now:
+                        expand = false;
+                    }
+                }
+                else
+                {
+                    //There are no vertices left to expand
+                    expand = false;
+                }
+
+
+//                //Prune back any edges in the front that have previously failed:
+//                if (useFailureTracking_ == true)
+//                {
+//                    if (edgeQueue_.begin()->second.first->hasAlreadyFailed(edgeQueue_.begin()->second.second) == true)
+//                    {
+//                        //Remove the edge from the queue:
```

Comment:
Please remove comments. There is also quite some white space between functions. We usually use just one empty line.